server-phoenix/lib/helios_web/controllers/events_api/events_api_controller.ex

```diff
@@ -0,0 +1,55 @@
+defmodule HeliosWeb.EventsAPI.EventsAPIController do
+  use HeliosWeb, :controller
+  alias Helios.{Repo, Event}
+
+  def handle(conn, params) do
+    cond do
+      params["version"] == "1" -> api_v1(conn, params)
+      true -> send_resp(conn, 404, "Not Found")
+    end
+  end
+
+  defp api_v1(conn, params) do
+    case params["event_type"] do
+      "github_commit" ->
+        post_event(
+          conn,
+          &Event.commits/1,
+          :github_commit,
+          params["event_id"],
+          params["author"]
+        )
+
+      _ ->
+        send_resp(conn, 400, "Bad Request")
+    end
+  end
+
+  defp post_event(conn, get_all_of_source_type, source, external_id, source_author) do
+    unless(
+      Event
+      |> get_all_of_source_type.()
```

Comment:
You can just pass in the Ecto Query from `Event.commits(Event)`:
```
post_event(
  conn,
  Event.commits(Event),
  ....

  defp post_event(conn, source_type_query, source, external_id, source_author) do
    ...
    unless(
      source_type_query
      |> Event.with_external_id(external_id)
      ...
  ```

Reply:
updated!